### PR TITLE
Use Hermitian inner product and reduce number of tests

### PR DESCRIPTION
**Pull request by sethaxen**
Following https://github.com/JuliaNLSolvers/ManifoldsBase.jl/pull/18, don't constrain inner products for complex manifolds to be real.

### CODE REVIEW

File: test/grassmann.jl [hunk 1]
```diff
@@ -85,7 +89,7 @@ include("utils.jl")
                 test_reverse_diff = false,
                 retraction_methods = [PolarRetraction(), QRRetraction()],
                 inverse_retraction_methods = [PolarInverseRetraction(), QRInverseRetraction()],
-                exp_log_atol_multiplier = 10.0,
+                exp_log_atol_multiplier = 5000.0,
```
Comment (mateuszbaran) [inline test/grassmann.jl]: Could you set a lower multiplier for `ComplexF64`?

**Comment by sethaxen** [comment]
Any tips on how to fix the failing x86 Windows builds? Even the Float64 checks fail there, and the tolerances look really low still.

**Comment by mateuszbaran** [comment]
These Windows x86 builds have tolerance issues in many tests and slow down AppVeyor CI quite a bit. Maybe we should just get rid of them? I don't think there is anyone using 32-bit Windows for numerical computing.

**Comment by sethaxen** [comment]
For Julia 1, it looks like we're sometimes exceeding the maximum time limit of 50 minutes. There are a few ways we can handle this:
- Remove some test cases
- Don't test against Julia 1
- Permit failures against Julia 1
- Split tests into different stages
- Other solutions?

@kellertuer @mateuszbaran what are your thoughts?

**Comment by mateuszbaran** [comment]
The time limit issue doesn't look good and will only get worse with more manifolds. I think we should keep testing against Julia 1.0 for now. What do you mean by splitting tests into different stages?

For now, the easiest solution would be removing `Float32` `MArray` tests, `StaticArrays` compiles slowly and the `Float32` case isn't that useful. Long term, we may have to split the package similarly to `DifferentialEquations`.

**Comment by kellertuer** [comment]
How would that splitting work? We would do `BasicManifolds.jl` and such? Should we then found our own Organisation (JuliaManifolds or something)?

In general I think we don't have to check all combinations, so if it works for `Float64`, `Float32` and `MArray{Float64}` I'd be fine (similarly for other decorator-style types).

**Comment by sethaxen** [comment]
> The time limit issue doesn't look good and will only get worse with more manifolds. I think we should keep testing against Julia 1.0 for now. What do you mean by splitting tests into different stages?

We currently have 2 build stages: tests and documentation. We can further split the tests into several groups; each group would have its own stage (see https://docs.travis-ci.com/user/build-stages/define-steps/). The downside here is that the total wallclock time for testing and the number of jobs will go up.

I'm fine with reducing the number of array/number types that we test against. It's a band-aid, but a good one for now.

**Comment by mateuszbaran** [comment]
Yes, something like `BasicManifolds`, `GroupManifolds` etc. I don't recommend rushing it, coordinating changes across many packages is more difficult than working in a single repository. `ManifoldsBase.jl` is already separated which will help.

Removing unnecessary tests is much easier than making new packages.

> Should we then found our own Organisation (JuliaManifolds or something)?

Maybe? JuliaNLSolvers maintainers keep setting up everything we need though.

> We currently have 2 build stages: tests and documentation. We can further split the tests into several groups; each group would have its own stage (see https://docs.travis-ci.com/user/build-stages/define-steps/). The downside here is that the total wallclock time for testing and the number of jobs will go up.

Nice, I didn't know about this. BTW, I think we can remove Julia 1.1 to reduce the test time. Only the latest stable and LTS releases of Julia are officially supported and that's 1.0 and 1.3 currently.

**Comment by kellertuer** [comment]
Yes, let's not rush. For more than the current 2 packages, I would be in favour of an own organisation, I am sure we can manage to set that up, too. 

And of course, it's simpler to stay with 2 packages as long as we can, and moving a few parts over to base would also be an idea, as soon as certain parts of the interface are more stable.

**Comment by sethaxen** [comment]
Travis's OS X builds seem to hang in the queue today. When they finish running, I'll merge this PR unless there are any objections to the reductions I've made to the tests.

**Pull request merged**